### PR TITLE
Glob_Patterns.rst/Install.rst/Tutorial.rst: Fix external links which point to internal pages

### PR DESCRIPTION
**Pull request by Gromy1211**
This changes direct links to relative links to the internal page.

Closes https://github.com/coala/documentation/issues/595

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: Users/Tutorial.rst [hunk 1]
```diff
@@ -42,7 +42,7 @@ Command Line Interface
 
 In order to specify the files to analyze, you can use the ``--files``
 argument of coala like demonstrated below. For all file paths, you can
-specify (recursive) `globs <../Users/Glob_Patterns.html>`__.
+specify (recursive) :doc:`globs <Glob_Patterns>`  .
```
Comment (li-boxuan) [inline Users/Tutorial.rst]: Redundant whitespace at the end of the sentence.

File: Users/Glob_Patterns.rst [hunk 2]
```diff
@@ -13,8 +13,7 @@ Suppose you want ``SpaceConsistencyBear`` to perform an analysis on a file
 .. note::
 
     If you don't know the functions of a bear or how to perform the analysis
-    with a bear, you should go through `Tutorial
-    <http://coala.readthedocs.io/en/latest/Users/Tutorial.html>`_ first.
+    with a bear, you should go through :doc:`Tutorial <Tutorial>`  first.
```
Comment (areebbeigh) [inline Users/Glob_Patterns.rst]:
```suggestion
    with a bear, you should go through :doc:`Tutorial <Tutorial>` first.
```

Comment (KVGarg) [reply Users/Glob_Patterns.rst]: Also, `the Tutorial` will look better, giving more importance to the word "tutorial"

File: Users/Install.rst [hunk 3]
```diff
@@ -341,7 +341,7 @@ Shell-Autocompletion Support
 ----------------------------
 
 If you are a bash/zsh user , checkout the
-`guide <https://docs.coala.io/en/latest/Users/Shell_Autocompletion.html>`_
+:doc:`guide <Shell_Autocompletion>`
```
[the remark this answers is not in the document]
Comment (KVGarg) [reply Users/Install.rst]: @Gromy1211 There is an extra space at the end of line. Remove it to get you PR error free

**Comment by KVGarg** [comment]
Also, you need to [squash](url
http://api.coala.io/en/latest/Developers/Git_Basics.htm#squashing_your_commitsl) your commits.

File: Users/Install.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (Gromy1211) [reply Users/Install.rst]: image
I think I have already removed the space, I don't know why gitmate-bot mentions it.

[the remark this answers is not in the document]
Comment (KVGarg) [reply Users/Install.rst]: Ok, will see this. Ok Please squash  your commits as well as make your heading small of upto 50 characters

[the remark this answers is not in the document]
Comment (KVGarg) [reply Users/Install.rst]: Checkout [comment](url
https://github.com/coala/documentation/pull/596#pullrequestreview-202527696)

[the remark this answers is not in the document]
Comment (Gromy1211) [reply Users/Install.rst]: Thanks a lot! I am working on the commits ;)

**Comment by jayvdb** [comment]
ack https://github.com/coala/documentation/pull/596/commits/218b9d3c0e870a7586406315d46f3ba4b641931b

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**